test/collect.cc

```diff
@@ -39,5 +39,35 @@ TEST(Collect, CommonSetPass) {
   EXPECT_EQ(12, *++result.begin());
 }
 
+TEST(Collect, VectorToRepeatedPtrField) {
+  std::vector<std::string> v = {"Hello", "World"};
+
+  auto s = [&]() {
+    return Iterate(v)
+        | Collect<google::protobuf::RepeatedPtrField<std::string>>();
+  };
+
+  google::protobuf::RepeatedPtrField<std::string> result = *s();
+
+  EXPECT_EQ(result.size(), 2);
+  EXPECT_STREQ("Hello", result.begin()->c_str());
```

Comment:
Whenever you're passing `foo->c_str()` to `EXPECT_STREQ`, use `EXPECT_EQ` instead:

`EXPECT_EQ("Hello", *result.begin());`

Here and below.

See discussion: https://github.com/3rdparty/stout/pull/53#discussion_r903157994